### PR TITLE
Add CPU count for VxWorks

### PR DESCRIPTION
**Pull request by kuhlenough**
OS specific API required

### CODE REVIEW

File: src/pthread/thread.cpp [hunk 1]
```diff
@@ -556,6 +560,18 @@ namespace boost
...
+        cpuset_t set =  ::vxCpuEnabledGet();
+  #ifdef __DCC__
+        int i;
+        for( i = 0; set; ++i)
+        {
+           set &= set -1;
+        }
+        return(i);
+  #else  
+        return (__builtin_popcount(set) );
```
Comment (viboes) [inline src/pthread/thread.cpp]: Does it means that when we use VxWorks, we have always a compiler that has this intrinsic?
With with compiler has you tested?

Comment (kuhlenough) [reply src/pthread/thread.cpp]: VxWorks is packaged with one of 4 validated compilers depending on the board support package and processor, so I know that Intel ICC, GCC, and llvm/clang have __builtin_popcount() and  Diab compiler (__DCC_) does not.  I didn't find a Boost macro for popcount() but I agree its a cleaner solution.

Comment (viboes) [reply src/pthread/thread.cpp]: Okay, I accept the PR. We will adapt if someone else need something else.

Comment (viboes) [reply src/pthread/thread.cpp]: Thanks for the PR.

**Pull request merged**